### PR TITLE
Make Stamps deterministic

### PR DESCRIPTION
**Pull request by jjudd**

### CODE REVIEW

File: src/main/scala/higherkindness/rules_scala/workers/zinc/compile/AnalysisStore.scala [hunk 1]
```diff
@@ -690,10 +690,38 @@ class AnxAnalyses(format: AnxAnalysisStore.Format) {
...
+      .putAllBinaryStamps(sortedBinaryStamps)
+      .clearSourceStamps()
+      .putAllSourceStamps(sortedSourceStamps)
+      .build()
+  }
+
+  def sortStampTypeMap(
+    stampTypeMap: util.Map[String, Schema.Stamps.StampType]
+  ): TreeMap[String, Schema.Stamps.StampType] = {
+    TreeMap[String, Schema.Stamps.StampType]() ++ stampTypeMap.asScala
```
Comment (JaredNeil) [inline src/main/scala/higherkindness/rules_scala/workers/zinc/compile/AnalysisStore.scala]:
```suggestion
    TreeMap.from(stampTypeMap.asScala)
```
Though going from a Java type to a Scala type just to put it ordered map and then convert back to a Java type seems wasteful. Why not go directly to the ordered Java map?

**Review by JaredNeil — CHANGES_REQUESTED** [review]
Do you have any performance numbers on this change? I know there was a different change that made things much slower at some point, but I don't remember if it was similar to this in any way.

File: src/main/scala/higherkindness/rules_scala/workers/zinc/compile/AnalysisStore.scala (hunk 1 above)
Comment (jjudd) [reply src/main/scala/higherkindness/rules_scala/workers/zinc/compile/AnalysisStore.scala]: Good questions, I don't remember this code was written so long ago. Lemme go take another look. I bet we can go directly to the ordered Java map and that was a rough edge I forgot about last year.

**Comment by jjudd** [comment]
The performance regression last year wasn't caused by the sorting to make things deterministic. It was caused by a change I made in the worker that worked around the Scala 2 compiler tasty classloader caching bug before I knew that bug existed. I don't remember exactly the details of that regression as it's been too long, but I know it wasn't related to the sorting changes like these.

File: src/main/scala/higherkindness/rules_scala/workers/zinc/compile/AnalysisStore.scala (hunk 1 above)
Comment (jjudd) [reply src/main/scala/higherkindness/rules_scala/workers/zinc/compile/AnalysisStore.scala]: I updated this to use a `java.util.TreeMap` directly.

**Review by JaredNeil — APPROVED** [review]
Sure would be nice if they just used `TreeMap` internally from the beginning.

**Pull request merged**